### PR TITLE
Fix crash introduced in v0.10.8 and slightly improve loader

### PR DESCRIPTION
**Pull request by gkalpak**
This PR includes:

- A fix for the crash introduced in v0.10.8.
- Some small improvement to the loader:
  * Including some not-yet-released `minErr`-related fixes (see angular/angular.js#15881).
  * Less intrusive modifications affecting v1.2 apps only, not all apps. ("Accidentally" fixes #88.)
- Some random minor refactorings (done while debugging the crash).

Fixes #88, #310.

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -101,8 +109,7 @@ function bufferData(tabId, message) {
 // context script –> background
 chrome.runtime.onMessage.addListener(brokerMessage);
 
-chrome.runtime.onConnect.addListener(function(devToolsPort) {
-
+chrome.runtime.onConnect.addListener(function (devToolsPort) {
```
Comment (petebacondarwin) [inline background.js]: what is it with adding spaces after `function`?

File: background.js [hunk 2]
```diff
@@ -71,13 +72,20 @@ function bufferData(tabId, message) {
   }
 
   if (message.event === 'scope:new') {
-    tabData.scopes[message.data.child] = {
```
Comment (petebacondarwin) [inline background.js]: I note that this change no longer modifies the tabData object. Is that on purpose?

Comment (petebacondarwin) [inline background.js]: I note that this change no longer modifies the `tabData.scope` object. Is that on purpose?

File: background.js (hunk 1 above)
Comment (gkalpak) [inline background.js]: It falls under the random changes category :smiley:
Both variations are used throughout the project - it drives me crazy. I was trying to blend in with the rest of the file :grin:

File: background.js (hunk 2 above)
Comment (gkalpak) [inline background.js]: It does. I just stored some properties as local var (`childScopeData = tabData.scopes[childId]`).

Comment (petebacondarwin) [inline background.js]: At no point do you write to `tabData.scopes[message.data.child]` any more

Comment (petebacondarwin) [inline background.js]: the `childScopeData` never gets assigned onto the `tabData.scopes` object.

File: hint.js [hunk 3]
```diff
@@ -5,12 +5,20 @@
...
 require('angular-hint');
+
+angular.hint.onAny(function (event, data, severity) {
+  // EventEmitter2 usually omits the event type for the argument list (assigning it to `this.event`
+  // instead), but under certain circumstances it may include it.
+  if (this.event !== event) {
+    severity = data;
+    data = event;
+    event = this.event;
+  }
```
Comment (petebacondarwin) [inline hint.js]: wot no tests? :-P

File: background.js (hunk 2 above)
Comment (gkalpak) [inline background.js]: OMG, you are right :scream:

File: hint.js (hunk 3 above)
Comment (gkalpak) [inline hint.js]: I know, right? :P

File: background.js (hunk 2 above)
Comment (gkalpak) [inline background.js]: So, this was used to "hydrate" the Scopes view when the DevTools would open, at which point the DevTools app would take over maintaining the scopes info. With my incorrect change, the scopes would not be hydrated when you opened DevTools, so you needed to refresh the app with the DevTools open.

Thx for watching out for my blunders :smiley:
It really sucks that there are no substancial tests. Hoping to fix it some time.

**Comment by gkalpak** [comment]
Fixed the `scopes` thing in https://github.com/angular/batarang/pull/311/commits/68f8830e904b2427014385d852ae654b7f1a7d61. Thx!
Merging...

**Pull request closed without merging**